### PR TITLE
Implement first part of React Dom Server in std/types

### PR DESCRIPTION
**Pull request by Soremwar**
This PR implements types, tests and documentation for Facebook's react-dom-server library

### Note 
This PR relies in CDN React and ReactDom libraries for its tests
This PR also removes previous tests for ReactDom in order to replace them later for a full test suite

Ref #4722

### CODE REVIEW

File: std/types/react-dom/README.md [hunk 1]
```diff
@@ -11,3 +11,21 @@ Facebook's React library.
 // @deno-types="https://deno.land/std/types/react-dom/v16.13.1/react-dom.d.ts"
 import ReactDOM from "https://cdn.pika.dev/@pika/react-dom@v16.13.1";
 ```
+
+```typescript
+// @deno-types="https://deno.land/std/types/react-dom/v16.9.0/server/react-dom-server.d.ts"
```
Comment (axetroy) [inline std/types/react-dom/README.md:16]: I think the original file structure should be retained

the file should be `react-dom/server.d.ts`, not `react-dom-server.d.ts`

Comment (Soremwar) [reply std/types/react-dom/README.md:16]: Minor change. Will do it in a while

**Comment by ry** [comment]
I think these types should be pushed out of std. Can you register deno.land/x/types ?

**Comment by Soremwar** [comment]
@ry Will do

**Comment by Soremwar** [comment]
Closing in favor of https://github.com/denoland/deno_website2/pull/325

**Pull request closed without merging**